server/src/storage/types.rs

```diff
@@ -150,7 +132,279 @@ impl SqlType {
+                    },
+                    CompType::SEThan => {
+                        self.compare_byte_greater_than(val, val2).map(|x| !x)
+                    },
+                }
+            },
+
+            &SqlType::Char(_) => {
+                match comp {
+                    CompType::Equ => {
+                        self.compare_byte_for_equal(val, val2)
+                    },
+                    CompType::NEqu => {
+                        self.compare_byte_for_equal(val, val2).map(|x| !x)
+                    },
+                    CompType::GThan => {
+                        self.compare_byte_greater_than(val, val2)
+                    },
+                    CompType::SThan => {
+                        self.compare_byte_lesser_than(val, val2)
+                    },
+                    CompType::GEThan => {
+                        self.compare_byte_lesser_than(val, val2).map(|x| !x)
+                    },
+                    CompType::SEThan => {
+                        self.compare_byte_greater_than(val, val2).map(|x| !x)
+                    },
+                }
+            },
+        }
+    }
+    /// fn compares slices of u8 byte for byte and returns if both values are equal
+    /// returns boolean on success and Error when given values do not have the same size
+    fn compare_byte_for_equal(&self, val: &[u8], val2: &[u8])
+    -> Result<bool, Error>
+    {
+        if val.len() != val2.len() {
+            return Err(Error::WrongLength)
+        }
+        info!("start comparing each byte");
+        for i in 0 .. val.len() {
+            if val[i] != val2[i] {
+                return Ok(false)
+            }
+        }
+        Ok(true)
+    }
+    /// fn compares slices of u8 byte for byte and returns
+    /// if first given value is greater than the second one
+    /// returns boolean on success and Error when given values do not have the same size
+    fn compare_byte_greater_than(&self, val: &[u8], val2: &[u8])
+    -> Result<bool, Error>
+    {
+        info!("start comparing each byte");
+        if val.len() != val2.len() {
+            return Err(Error::WrongLength)
+        }
+        for i in 0 .. val.len() {
+            if val[i] > val2[i] {
+                return Ok(true)
+            }
+        }
+        Ok(false)
+    }
+
+    /// fn compares slices of u8 byte for byte and returns
+    /// if first given value is lesser than the second one
+    /// returns boolean on success and Error when given values do not have the same size
+    fn compare_byte_lesser_than(&self, val: &[u8], val2: &[u8])
+    -> Result<bool, Error>
+    {
+        info!("start comparing each byte");
+        if val.len() != val2.len() {
+            return Err(Error::WrongLength)
+        }
+        for i in 0 .. val.len() {
+            if val[i] < val2[i] {
+                return Ok(true)
+            }
+        }
+        Ok(false)
+    }
```

Comment:
Same thing with [`PartialOrd`](https://doc.rust-lang.org/std/cmp/trait.PartialOrd.html). You can use `<`, `<=`, `>=` and `>` I think...